src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerHelper.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+using Org.OData.Core.V1;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.AspNet.OData.Formatter.Serialization
+{
+    /// <summary>
+    /// Helper class for OData Serialization
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1053:StaticHolderTypesShouldNotHaveConstructors")]
+    public class ODataSerializerHelper
+    {
+        internal static void AppendInstanceAnnotations(ODataResourceBase resource, ResourceContext resourceContext, object value, ODataSerializerProvider SerializerProvider)
```

Comment:
>object [](http://example.com/codeflow?start=116&length=6)

You can save a cast within the method for by making this an IODataInstanceAnnotationContainer, and requiring that the caller pass as an IODataInstanceAnnotationContainer (since, in many cases, the caller already knows it is an IODataInstanceAnnotationContainer).  Keep the null check on line 22, though, so if the caller does pass in their object cast to an IODataInstanceAnnotationContainer, and the cast fails, it will be a no-op.

Reply:
This was kept so because from the calling method it sends an object(got from trygetvalue outs object) . So a cast would be inevitable, may be can move that cast to the calling method